### PR TITLE
Add Rhode Island 2023 rules 

### LINKED ISSUE #4197: Add 2023 Rhode Island Tax Rules
2023 Income Tax Form Instructions: https://tax.ri.gov/sites/g/files/xkgbur541/files/2023-12/2023%201040R%20Instructions.pdf

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #4197

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ri/tax/income/agi/subtractions/social_security/limit/income.yaml [hunk 1]
```diff
@@ -13,13 +13,16 @@ metadata:
   # The tax form points to this worksheet: Social Security modification on Schedule M, line 1t.
   - title: Rhode Island 2022 Modification for Taxable social security income worksheet
     href: https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-12/Social%20Security%20Worksheet_w.pdf#page=1
+  - title: Rhode Island 2023 Modification for Taxable social security income worksheet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/agi/subtractions/social_security/limit/income.yaml:16]: I think the amounts here are adjusted with inflation:
image

We need to add some metadata for inflation adjustments and rounding, see DC:
https://github.com/PolicyEngine/policyengine-us/pull/4306

File: policyengine_us/parameters/gov/states/ri/tax/income/credits/property_tax/max_amount.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ values:
   2020-01-01: 400
   2021-01-01: 415
   2022-01-01: 600
+  2023-01-01: 650
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/credits/property_tax/max_amount.yaml:19]: could you add uprating and rounding to the property tax files as well?

File: policyengine_us/parameters/gov/states/ri/tax/income/exemption/amount.yaml [hunk 3]
```diff
@@ -2,6 +2,7 @@ description: Rhode Island provides the following amount for each exemption in th
 values:
   2021-01-01: 4_250
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/exemption/amount.yaml:3]: could you check all the file that adjust every year if we need updates and rounding?

File: policyengine_us/parameters/gov/states/ri/tax/income/rate.yaml [hunk 4]
```diff
@@ -14,16 +16,24 @@ metadata:
...
       2021-01-01: 0.0375
   - threshold:
-      2021-01-01: 66_200
-      2022-01-01: 68_200
+      values:
+        2021-01-01: 66_200
+        2022-01-01: 68_200
+        2023-01-01: 73_450
+      metadata:
+        uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/rate.yaml:28]: might need rounding here too right? Maybe nearest 50

File: policyengine_us/parameters/gov/states/ri/tax/income/deductions/standard/phase_out/increment.yaml [hunk 5]
```diff
@@ -2,6 +2,7 @@ description: Rhode Island phases the standard deduction out at this increment.
 values:
   2021-01-01: 6_000
   2022-01-01: 6_200
+  2023-01-01: 6_700
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/deductions/standard/phase_out/increment.yaml:5]: I think we need to uprate this as well:
image

File: policyengine_us/parameters/gov/states/ri/tax/income/rate.yaml [hunk 6]
```diff
@@ -5,7 +5,14 @@ metadata:
   rate_unit: /1
   period: year
   label: Rhode Island income tax rate
+  uprating:
+    parameter: gov.irs.uprating
+    rounding:
+      type: downwards
+      interval: 50
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ri/tax/income/rate.yaml:12]: we need to add this metadata piece to each individual threshold (besides 0) as this structure will also uprate the rate, which should stay the same

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**